File: 16-express-js-hexa/adapters/persistence/dpos/userDPO.js
import { User } from '../../../domain/models/user.js'

export class UserDPO {
    constructor(id, name, email, passwordHash, role, lastLogin) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.password_hash = passwordHash; // Angepasst an DB-Konvention
        this.user_role = role;            // Angepasst an DB-Konvention
        this.last_login = lastLogin;      // Angepasst an DB-Konvention
    }

    static fromDomain(user) {
        return new UserDPO(
            user.id,
            user.name,
            user.email,
            user.passwordHash,
            user.role,
            user.lastLogin
        );
    }

    static toDomain(dpo) {
        return new User(
            dpo.id,
            dpo.name,
            dpo.email,
            dpo.password_hash,
            dpo.user_role,
            dpo.last_login
        );
    }
}

File: 16b-fastapi-python-hexa/adapters/persistence/dpos/user_dpo.py
from dataclasses import dataclass
from domain.models.user import User
from datetime import datetime
from typing import Optional

@dataclass
class UserDPO:
    id: str
    name: str
    email: str
    password_hash: str
    user_role: str
    last_login: Optional[datetime]

    @classmethod
    def from_domain(cls, user: User) -> 'UserDPO':
        return cls(
            id=user.id,
            name=user.name,
            email=user.email,
            password_hash=user.password_hash,
            user_role=user.role,
            last_login=user.last_login
        )

    def to_domain(self) -> User:
        return User(
            id=self.id,
            name=self.name,
            email=self.email,
            password_hash=self.password_hash,
            role=self.user_role,
            last_login=self.last_login
        )